+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                                    lambda_tangent_ijk = obj.w.lambda_tangent(ii,jj,kk);
+                                    delta_d_ijk = obj.w.delta_d(ii,jj,kk);
+                                    gamma_d_ijk = obj.w.gamma_d(ii,jj,kk);
+
+                                    Gij = vertcat(Gij, {lambda_tangent_ijk});
+                                    Hij = vertcat(Hij, {delta_d_ijk});
+                                    Gij = vertcat(Gij, {beta_ijk});
+                                    Hij = vertcat(Hij, {gamma_ijk});
+                                    
+                                  case 'Conic'
+                                        beta_ijk = obj.w.beta(ii,jj,kk);
+                                        gamma_ijr = obj.w.gamma(ii,jj,kk);
+                                        
+                                        Gij = vertcat(Gij, {beta_ijk});
+                                        Hij = vertcat(Hij, {gamma_ijk});
+                                    switch opts.conic_model_switch_handling
+                                      case 'Plain'
+                                        % no extra expr
+                                      case 'Abs'
+                                        p_vt_ijk = obj.w.p_vt(ii,jj,kk);
+                                        n_vt_ijk = obj.w.n_vt(ii,jj,kk);
+                                        
+                                        Gij = vertcat(Gij, {p_vt_ijk});
+                                        Hij = vertcat(Hij, {n_vt_ijk});
+                                      case 'Lp'
+                                        alpha_vt_ijk = obj.w.alpha_vt(ii,jj,kk);
+                                        p_vt_ijk = obj.w.p_vt(ii,jj,kk);
+                                        n_vt_ijk = obj.w.n_vt(ii,jj,kk);
+                                        
+                                        Gij = vertcat(Gij, {alpha_vt_ijk});
+                                        Hij = vertcat(Hij, {p_vt_ijk});
+                                        Gij = vertcat(Gij, {1-alpha_vt_ijk});
+                                        Hij = vertcat(Hij, {n_vt_ijk});
+                                    end
+                                end
+                            end
+                            obj.G.standard_comp(ii,jj,kk) = {vertcat(Gij{:})};
+                            obj.H.standard_comp(ii,jj,kk) = {vertcat(Hij{:})};
+                        end
+                    end
+                end
+            end
+        end
+
+        function generate_step_equilibration_constraints(obj)
+            import casadi.*
+            model = obj.model;
+            opts = obj.opts;
+            dims = obj.dcs.dims;
+            
+            if ~opts.use_fesd % do nothing
+                return
+            end
+            rbp = ~opts.right_boundary_point_explicit;
+            % TODO (@anton) maybe unify some of these calculations
```

Comment:
this would be nice, as there is much repetition. and we may drop support for direct (and even direct homotopy?) as they are not the choice to be used in practice.

Reply:
I think this should be done in a pass that resolves #111 but yes, agreed.